### ISSUE TITLE
ignoreSigs does not match generic functions

### ISSUE THREAD

**Issue by SOF3** [tomarrell/wrapcheck#49]
```go
func Wrapf[T any]() error { return nil }
```

```yaml
ignoreSigs:
  - Wrapf(
```

This function is not excluded, probably because wrapcheck matches `Wrapf[T any]()` instead of `Wrapf()`. This should be allowed, or at least documented in the readme.

**Comment by tomarrell** [comment]
Hey @SOF3, thanks for the report. 

One question, what happens when you use the config with signature:

```yaml
ignoreSigs:
  - Wrapf
```

Does this ignore the generic?

**Comment by SOF3** [comment]
in that case, yes it works since it is a substring

**Comment by tomarrell** [comment]
Glad to hear, so that solves your issue? You can define sigs `Wrapf(` and `Wrapf[` which will cover both generic and non-generic methods. 

Which package are you using these functions from?

**Comment by SOF3** [comment]
It is my internal package. I have a function like this:

```go
func TryMapSlice[T, U any](
    slice []T,
    mapFn func(T) (U, error),
) ([]U, error)
```

Usually the errors from an external package are already wrapped in `mapFn`, and `TryMapSlice` just further wraps it to indicate the index at which error occurred). There is no point to wrap it the third time.

```go
var foo []Foo = ...
bar, err := TryMapSlice(foo, func(foo Foo) (Bar, error) {
    bar, err := externalpackage.Transform(foo)
    if err != nil {
        return nil, Wrap(err, ...)
    }

    return bar, nil
})
```

There is no point in wrapping the outer `err` once again.

**Comment by tomarrell** [comment]
Ok I see, yes that's a good example for just a comment ignore in that situation. If you're using golangci-lint you can just throw in a `//nolint:wrapcheck` in there and it'll ignore. 

I'm going to consider this closed now, as I don't see it as an issue with the linter which can't be solved with existing configuration options. Let me know if you have any further questions!

**Comment by SOF3** [comment]
yeah it was just confusing to figure out why it wasn't ignored, would appreciate some clarification in documentation of the configuration.
